src/main.rs

```diff
@@ -52,6 +57,20 @@ fn open_dit_repo() -> Result<Repository> {
 }
 
 
+/// Get a vector of commits from values
+///
+/// This function transforms values to a vector.
+///
+fn values_to_hashes<'repo>(repo: &'repo Repository, values: Values) -> Result<Vec<Commit<'repo>>> {
+    let mut retval = Vec::new();
+    for commit in values.map(|string| repo.revparse_single(string))
+                        .map(|oid| repo.find_commit(try!(oid).id())) {
+        retval.push(try!(commit));
+    }
+    Ok(retval)
+}
```

Comment:
Not collecting isn't an option:

 * We need to know in advance whether an error occurred or not.
 * The commits have to be stored somewhere anyways.